WalletWasabi/Crypto/Api/ICredentialRequestBuilder.cs

```diff
@@ -0,0 +1,14 @@
+using NBitcoin;
+using WalletWasabi.Crypto.ZeroKnowledge;
+
+namespace WalletWasabi.Crypto.Api
+{
+	public interface ICredentialRequestBuilder
```

Comment:
why is this interface needed?

Reply:
This is needed in order to allow a same type have more than one interface. It is just for API usability.

Reply:
so basically just for clarity/readability, and to allow the client class to be refactored more easily at a later point if we need to?

my gut feeling is usually to do this kind of thing, but over time i learned to kind of suppress that instinct, since it's usually better to extract interfaces a bit later (when polymorphism is actually necessary)